queue/src/fiskaltrust.Interface.Tagging/ReceiptConverter.cs

```diff
@@ -0,0 +1,79 @@
+﻿using fiskaltrust.ifPOS.v1;
+using fiskaltrust.Interface.Tagging.Interfaces;
+using fiskaltrust.Interface.Tagging.ErrorHandling;
+using fiskaltrust.Interface.Tagging.Models.Extensions;
+namespace fiskaltrust.Interface.Tagging
+{
+    public class ReceiptConverter
+    {
+        private readonly ICaseConverterFactory _caseConverterFactory;
+        public ReceiptConverter()
+        {
+
+            _caseConverterFactory = new CaseConverterFactory();
+
+        }
+        public ReceiptRequest ConvertRequestToV1(ReceiptRequest request)
+        {
+            var converter = _caseConverterFactory.CreateInstance(request.ftReceiptCase);
+            if (!request.IsVersionV2())
+            {
+                throw new ReceiptCaseVersionException($"It's NOT a V2 receipt case. Found V{request.GetVersion()}.");
+            }
+            converter.ConvertftReceiptCaseToV1(request);
+
+            if (request.cbChargeItems != null)
+            {
+                foreach (var chargeItem in request.cbChargeItems)
+                {
+                    if (!chargeItem.IsVersionV2())
+                    {
+                        throw new ChargeItemVersionException($"It's NOT a V2 charge Item. Found V{chargeItem.GetVersion()}.");
+                    }
+                    converter.ConvertftChargeItemCaseToV1(chargeItem);
+                }
+            }
+
+            if (request.cbPayItems != null)
+            {
+                foreach (var payItem in request.cbPayItems)
+                {
+                    if (!payItem.IsVersionV2())
+                    {
+                        throw new PayItemVersionException($"It's NOT a V2 pay Item. Found V{payItem.GetVersion()}.");
```

Comment:
```suggestion
                        throw new PayItemVersionException($"Using the v2 tagging in ftReceiptCases requires the usage of v2 PayItems (mixing versions within requests is not possible). Found v{payItem.GetVersion()} instead of v2.");
```